lenskit-core/src/main/java/org/lenskit/util/io/ObjectStreams.java

```diff
@@ -159,18 +161,22 @@ public void close() {
      * @return A new list containing the elements of the object stream.
      */
     @SuppressWarnings("PMD.LooseCoupling")
-    public static <T> ArrayList<T> makeList(@WillClose ObjectStream<? extends T> objectStream) {
-        ArrayList<T> list;
+    public static <T> List<T> makeList(@WillClose ObjectStream<? extends T> objectStream) {
```

Comment:
This is the heart of some object stream changes. What we're doing is enabling an object stream that wraps a list to be *unwrapped*, instead of requiring the copy all the time, to fast-path some common operations.